src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -627,6 +702,34 @@ public int getWorstCaseFrequencyDepartureTime (TripSchedule schedule, int stopPo
         return Math.max(latestBoardTimeAssumingEntryIsAlreadyRunning, latestBoardTimeOfFirstTrip);
     }
 
+    /**
+     * For half-headway (non-monte-carlo) evaluation of frequency-based routes.
+     */
+    public static int getAverageCaseFrequencyDepartureTime (
+            TripSchedule schedule,
+            int stopPositionInPattern,
+            int frequencyEntryIdx,
+            int earliestTime
+    ) {
+        int headway = schedule.headwaySeconds[frequencyEntryIdx];
+        int halfHeadway = headway / 2;
+
+        // Ensure the schedule is still running.
+        int travelTimeFromStartOfTrip = schedule.departures[stopPositionInPattern];
+
+        // Not strictly correct, should be revised: the last vehicle could leave the terminal as early as headwaySeconds
+        // before the end of the frequency entry. We might want to take the minimum or average of waiting for the
+        // current entry, and waiting for the next entry. We need to exclude entries that have ended, but not entries
+        // that have not yet started (because you can wait for them to start).
```

Comment:
Is https://github.com/conveyal/r5/issues/122 relevant here?

Reply:
Yes. I added that reference to the comment.